File: src/test/resources/support/handler_invoker.rb
module Support
  class HandlerInvoker
    def self.invoke_string_handler
      yield 'callback_value'
    end
    def self.invoke_string_handler_first_param handler, other
      handler.call other
    end
    def self.invoke_string_handler_last_param other
      yield other
    end
    def self.invoke_async_result_handler_success(&callback)
      callback.call nil, 'hello'
    end
    def self.invoke_async_result_handler_failure(&callback)
      future = Java::IoVertxCore::Future.failedFuture 'oh no'
      callback.call future.cause, nil
    end
  end
end

File: src/test/resources/io/vertx/scala/support/HandlerInvoker.scala
package io.vertx.scala.support

import java.lang.Exception
import scala.util.{Failure, Success, Try}

object HandlerInvoker {

  def invokeStringHandler(handler:String => Unit) {
    handler("callback_value")
  }

  def invokeStringHandlerFirstParam(handler:String => Unit, other: String) {
    handler(other)
  }

  def invokeStringHandlerLastParam(other: String, handler:String => Unit) {
    handler(other)
  }

  def invokeAsyncResultHandlerSuccess(f: PartialFunction[Try[String], Unit]) {
    f(Success("hello"))
  }

  def invokeAsyncResultHandlerFailure(f: PartialFunction[Try[String], Unit]) {
    f(Failure(new Exception("oh no")))
  }
}